RNF34.py

```diff
@@ -0,0 +1,134 @@
+from abc import ABC, abstractmethod
+
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __init__(self, sequence):
+        pass
+
+    @abstractmethod
+    def __len__(self):
+        pass
+
+    @abstractmethod
+    def __getitem__(self, slc):
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    @abstractmethod
+    def __repr__(self):
+        pass
+
+    @abstractmethod
+    def is_valid_alphabet(self):
+        pass
+
+
+class NucleicAcid(BiologicalSequence):
+    def __init__(self, sequence):
+        self.sequence = sequence
+
+    def __len__(self):
+        return len(self.sequence)
+
+    def __getitem__(self, slc):
+        return self.sequence[slc]
+
+    def __str__(self):
+        return str(self.sequence)
+
+    def __repr__(self):
+        return self.sequence
```

Comment:
В методе repr мы должны скорее передавать техническую информацию о классе. Я тоже в своей дз передала self.sequence, но при ревью подумала, что правильнее было бы использовать строку c указанием класса.
```suggestion
        return f'NucleicAcid("{self.sequence}")'
```

Reply:
Верно подмечено!